src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/pureThermalDiffusion/pureThermalDiffusionAroundWellbore.py

```diff
@@ -0,0 +1,128 @@
+	 
+def computeThermalDiffusionCoefficient(thermalConductivity, volumetricHeatCapacity):
+	return thermalConductivity / volumetricHeatCapacity
+	
+def extractDataFromXMLList(paramList):
+	# Extract data from a list in XML such as "{ 1, 2, 3}"
+	return paramList.replace('{', '').replace('}', '').strip().split(',')
+
+def getWellboreGeometryFromXML(xmlFilePath):
+	tree = ElementTree.parse(xmlFilePath)
+
+	meshParam = tree.find('Mesh/InternalWellbore')
+	radii = extractDataFromXMLList( meshParam.get("radius") )
+
+	Rin = float(radii[0])
+	Rout = float(radii[-1])
+
+	return [Rin, Rout]
+
+def getLoadingFromXML(xmlFilePath):
+	tree = ElementTree.parse(xmlFilePath)
+	fsParams = tree.findall('FieldSpecifications/FieldSpecification')
+
+	for fsParam in fsParams:
+		if ( (fsParam.get('fieldName') == "pressure") & (fsParam.get('initialCondition') != "1") ): 
+			if fsParam.get('setNames') == "{ rneg }":
+				Pin = float(fsParam.get('scale'))
+			if fsParam.get('setNames') == "{ rpos }":
+				Pout = float(fsParam.get('scale'))
+
+	for fsParam in fsParams:
+		if ( (fsParam.get('fieldName') == "temperature") & (fsParam.get('initialCondition') != "1") ): 
+			if fsParam.get('setNames') == "{ rneg }":
+				Tin = float(fsParam.get('scale'))
+			if fsParam.get('setNames') == "{ rpos }":
+				Tout = float(fsParam.get('scale'))
+            
+	thermalConductivity = float( extractDataFromXMLList( tree.find('Constitutive/SinglePhaseConstantThermalConductivity').get('thermalConductivityComponents') )[0] )
+
+	volumetricHeatCapacity = float( tree.find('Constitutive/SolidInternalEnergy').get('volumetricHeatCapacity') )
+	
+	permeability = float( extractDataFromXMLList( tree.find('Constitutive/ConstantPermeability').get('permeabilityComponents') )[0] )
+
+	porosity = float( tree.find('Constitutive/PressurePorosity').get('defaultReferencePorosity') )
+	
+	fluidViscosity = float( tree.find('Constitutive/ThermalCompressibleSinglePhaseFluid').get('defaultViscosity') )
+
+	fluidCompressibility = float( tree.find('Constitutive/ThermalCompressibleSinglePhaseFluid').get('compressibility') )
+
+	fluidThermalExpansionCoefficient = float( tree.find('Constitutive/ThermalCompressibleSinglePhaseFluid').get('thermalExpansionCoeff') )
+
+	return [Pin, Pout, Tin, Tout, thermalConductivity, volumetricHeatCapacity, permeability, porosity, fluidViscosity, fluidCompressibility, fluidThermalExpansionCoefficient]
+
+
+def main():
+
+	xmlFilePath = "../../../../../../../inputFiles/singlePhaseFlow/thermalCompressible_2d"
+	
+	Rin, Rout = getWellboreGeometryFromXML(xmlFilePath+"_benchmark.xml")
+
+	Pin, Pout, Tin, Tout, thermalConductivity, volumetricHeatCapacity, permeability, porosity, fluidViscosity, fluidCompressibility, fluidThermalExpansionCoefficient = getLoadingFromXML(xmlFilePath+"_base.xml")
+
+	plt.figure(figsize=(10,7))
+	font = {'size'   : 16}
+	plt.rc('font', **font)
+
+	for chart_idx, idx in enumerate([1, 2, 5, 10]):
+		# Numerical results
+		data = pd.read_csv(f'data_{idx}.csv')
```

Comment:
Instead of csv files, can you use time history function (`PackCollection` tasks in xml file) to collect and output time history information in hdf5 files and postprocess these files with python script?